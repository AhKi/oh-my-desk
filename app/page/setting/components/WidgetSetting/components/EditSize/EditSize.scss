.EditSize {
  margin-top: 30px;
  border-top: 1px solid $color-gray300;
}

.EditSize__header {
  margin-top: 48px;
}

.EditSize-description {
  font-weight: 400;
  font-size: 13px;
  color: $color-middle-gray-blue;
  display: inline-block;
}

.EditSize__text-input {
  font-size: 15px;
  max-width: 126px;
  text-align: right;
  border: 1px solid $color-gray250;
  border-radius: 3px;
  color: $global-text-color;
  transition: all 0.2s;
  width: 100%;
  box-shadow: inset 1px 1px 3px 0 rgba(0,0,0,0.06);
  margin-right: 8px;
  background: $color-white;
  height: auto;
  padding: $unit * 1.5;

  &::placeholder {
    color: $color-gray-blue;
    opacity: 0.5;
  }

  &:disabled {
    background: $color-gray50;
    border: 1px solid $color-gray300;
    box-shadow: none;
    color: $color-gray500;
  }

  &:focus {
    border: 1px solid $color-light-gray-blue;
    outline: none;
  }
}

.EditSize__content {
  padding-left: 30px;
}