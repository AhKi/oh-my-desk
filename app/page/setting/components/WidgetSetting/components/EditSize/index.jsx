import React from 'react';
import PropTypes from 'prop-types';
import './EditSize.scss';

const propTypes = {
  item: PropTypes.shape({
    size: PropTypes.shape({
      height: PropTypes.number,
      width: PropTypes.number,
    }),
  }),
  onChangeHeight: PropTypes.func,
  onChangeWidth: PropTypes.func,
};
const defaultProps = {
  item: {},
  onChangeHeight() {},
  onChangeWidth() {},
};

class EditSize extends React.Component {
  render() {
    const { item, onChangeHeight, onChangeWidth } = this.props;

    return (
      <div className="EditSize space-6x">
        <p className="EditSize__header space-1x">
          <strong>
            Web widget size
          </strong>
        </p>
        <span className="EditSize-description space-4x">
          The size is preserved when the widget is switched off and on.
        </span>
        <div className="EditSize__content">
          <label htmlFor="width-input" className="InputSet__label">
            <span className="InputSet__label-text">
              Width
            </span>
            <input
              className="EditSize__text-input"
              id="width-input"
              type="number"
              value={item.size.width || ''}
              onChange={onChangeWidth}
            />
            px
          </label>
          <label htmlFor="height-input" className="InputSet__label">
            <span className="InputSet__label-text">
              Height
            </span>
            <input
              className="EditSize__text-input"
              id="height-input"
              type="number"
              value={item.size.height || ''}
              onChange={onChangeHeight}
            />
            px
          </label>
        </div>
      </div>
    );
  }
}

EditSize.propTypes = propTypes;
EditSize.defaultProps = defaultProps;

export default EditSize;
