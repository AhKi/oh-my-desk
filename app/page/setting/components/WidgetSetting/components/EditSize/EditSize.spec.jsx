import React from 'react';
import { shallow } from 'enzyme';

import EditSize from '.';

describe('<EditSize />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(
      <EditSize
        item={{
          size: {
            height: 100,
            width: 100,
          },
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match to snapshot when size is zero', () => {
    const wrapper = shallow(
      <EditSize
        item={{
          size: {
            height: 0,
            width: 0,
          },
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
