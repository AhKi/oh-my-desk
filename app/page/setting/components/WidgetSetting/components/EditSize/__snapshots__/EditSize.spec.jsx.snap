// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSize /> should match to snapshot when render default 1`] = `
<div
  className="EditSize space-6x"
>
  <p
    className="EditSize__header space-1x"
  >
    <strong>
      Web widget size
    </strong>
  </p>
  <span
    className="EditSize-description space-4x"
  >
    The size is preserved when the widget is switched off and on.
  </span>
  <div
    className="EditSize__content"
  >
    <label
      className="InputSet__label"
      htmlFor="width-input"
    >
      <span
        className="InputSet__label-text"
      >
        Width
      </span>
      <input
        className="EditSize__text-input"
        id="width-input"
        onChange={[Function]}
        type="number"
        value={100}
      />
      px
    </label>
    <label
      className="InputSet__label"
      htmlFor="height-input"
    >
      <span
        className="InputSet__label-text"
      >
        Height
      </span>
      <input
        className="EditSize__text-input"
        id="height-input"
        onChange={[Function]}
        type="number"
        value={100}
      />
      px
    </label>
  </div>
</div>
`;

exports[`<EditSize /> should match to snapshot when size is zero 1`] = `
<div
  className="EditSize space-6x"
>
  <p
    className="EditSize__header space-1x"
  >
    <strong>
      Web widget size
    </strong>
  </p>
  <span
    className="EditSize-description space-4x"
  >
    The size is preserved when the widget is switched off and on.
  </span>
  <div
    className="EditSize__content"
  >
    <label
      className="InputSet__label"
      htmlFor="width-input"
    >
      <span
        className="InputSet__label-text"
      >
        Width
      </span>
      <input
        className="EditSize__text-input"
        id="width-input"
        onChange={[Function]}
        type="number"
        value=""
      />
      px
    </label>
    <label
      className="InputSet__label"
      htmlFor="height-input"
    >
      <span
        className="InputSet__label-text"
      >
        Height
      </span>
      <input
        className="EditSize__text-input"
        id="height-input"
        onChange={[Function]}
        type="number"
        value=""
      />
      px
    </label>
  </div>
</div>
`;
